lib/media/streaming_engine.js

```diff
@@ -1718,6 +1720,8 @@ shaka.media.StreamingEngine.prototype.parseEMSG_ = function(
       messageData: messageData
     };
 
+    this.playerInterface_.onTimelineRegionAdded(emsg);
```

Comment:
This seems to be the wrong callback to use.  EmsgInfo is not a TimelineRegionInfo.

Reply:
If the definition of TimelineRegionInfo is changed so that TimelineRegionInfo#eventElement is nullable, you could create a TimelineRegionInfo out of an EmsgInfo. But I have no idea if this makes sense. I'm assuming that clients will be able to differentiate based on schemeIdUri, but this could also be a breaking change for clients that already depend on `timelineregion*` events for EventStream behaviour.

Can you advise?

Reply:
We have an 'emsg' event already, which is dispatched currently by StreamingEngine through playerInterface_.onEvent.  Perhaps you should move the responsibility for that event to somewhere else.  Timeline events are dispatched from PlayheadObserver today.